### ISSUE TITLE
numpy-devel or not?

### ISSUE THREAD

**Issue by sdebionne** [conda-forge/numpy-feedstock#230]
I get contradictory information from the documentation. From the [`run_exports`](https://conda-forge.org/docs/maintainer/pinning_deps.html#specifying-run-exports) section:

> The general idea is that the numpy-devel package should be used when a package is building against the C interface (i.e. it needs the compatibility bound), and the numpy package should be used when a package is using only the python interface.

From the [building against numpy](https://conda-forge.org/docs/maintainer/knowledge_base.html#building-against-numpy) section:

```
host:
  - numpy
run:
  - {{ pin_compatible('numpy') }}
```

It looks like only the `default` channel provides `numpy-devel`, so I guess the second option is the recommanded one?

**Comment by rgommers** [comment]
There's no such thing as `numpy-devel`. NumPy headers are inside the main package.

**Comment by sdebionne** [comment]
It's not a question of headers but `run_exports`, check [this thread](https://gitter.im/conda-forge/conda-forge.github.io?at=5b903eeeac25fd11b5e76fcd) for instance.

TLDR:

>Numpy is one that has two purposes: one being the python interface, and the other being the C interface. The latter has ABI concerns, the former does not.

> Having two packages is one way to communicate the different intent there we have numpy-devel that imposes run_exports (for communicating ABI compatibility), and numpy which does not both are actually metapackages; numpy-base has the "guts"

I currently build my extension package against `numpy-devel` from Anaconda, Inc. When the conda-forge channel is used, I get strange combinations like:

```
    numpy:            1.11.3-py37he5ce36f_1207   conda-forge                 
    numpy-base:       1.18.5-py37h2f8d375_0      defaults   
```

**Comment by rgommers** [comment]
> It's not a question of headers but `run_exports`, check [this thread](https://gitter.im/conda-forge/conda-forge.github.io?at=5b903eeeac25fd11b5e76fcd) for instance.

Ah right, thanks.

> It looks like only the `default` channel provides `numpy-devel`, so I guess the second option is the recommanded one?

Yes, all I've seen in conda-forge is `{{ pin_compatible('numpy') }}`, that's what SciPy, scikit-learn et al. use.

> I currently build my extension package against numpy-devel from Anaconda, Inc. When the conda-forge channel is used, I get strange combinations like:

You should not mix channels like that. The conda-forge docs recommend using strict channel priority, which should prevent some of that weirdness. But even so, if you need conda-forge packages, why not build against `numpy` from the `conda-forge` channel?

**Comment by sdebionne** [comment]
> The conda-forge docs recommend using strict channel priority.

Right, then I wonder why it's not set so in the `condaforge/linux-anvil-comp7` docker image?

>But even so, if you need conda-forge packages, why not build against numpy from the conda-forge channel?

We are indeed transitioning to conda-forge.

**Comment by h-vetinari** [comment]
As Ralf explained, there's no `numpy-devel` and the aspect about `numpy-base` should have been solved by #234. Closing.
